### PR TITLE
Update item model to include necessary attributes 96

### PR DESCRIPTION
**Pull request by scottalexandra**
Updated Items model to add location, unit size, unit weight, available, and shippable indicator
validated presence of location on item model
updated all of the item forms (new, edit)
updated items and supplier index views
updated item modal 
updated supplier item index so add to cart button no longer appears
updated attributes file and tests
updated seed data

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -33,4 +34,33 @@ def add_categories_to_item(categories)
...
+
+  def item_retired
+    if self.retired
+      "Retired"
+    else
+      "Active"
+    end
+  end
+
 end
```
Comment (laurawhalin) [inline app/models/item.rb:66]: This file is very easy to read. Good job!

File: app/views/partials/_edit_item.html.erb [hunk 2]
```diff
@@ -48,6 +69,20 @@
     </div>
 
     <div class="form-group">
+      <%= f.label :available, "Item Not Currently Available?", class: "col-sm-2 control-label" %>
```
Comment (laurawhalin) [inline app/views/partials/_edit_item.html.erb:72]: "Item Currently Available" would be easier to read at a glance, but you might need to change the default of the checkbox to fit the wording.

File: app/views/partials/_item_form.html.erb [hunk 3]
```diff
@@ -48,6 +69,20 @@
     </div>
 
     <div class="form-group">
+      <%= f.label :available, "Item Not Currently Available?", class: "col-sm-2 control-label" %>
```
Comment (laurawhalin) [inline app/views/partials/_item_form.html.erb:72]: Same comment here as above regarding the wording of the checkbox label.

File: app/views/partials/_new_item.html.erb [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+    <div class="form-group">
+      <%= f.label :retired, "Retire Item?", class: "col-sm-2 control-label" %>
+      <div class="col-sm-7">
+      <%= f.check_box :retired %>
+      </div>
+    </div>
+
+    <div class="form-group">
+      <%= f.label :available, "Item Not Currently Available?", class: "col-sm-2 control-label" %>
```
Comment (laurawhalin) [inline app/views/partials/_new_item.html.erb:72]: Same comment here about the wording of this checkbox.

File: app/models/item.rb [hunk 5]
```diff
@@ -4,6 +4,7 @@ class Item < ActiveRecord::Base
   validates :supplier_id, presence: true
   validates :price, presence: true, numericality: { :greater_than_or_equal_to => 0 }
 	validates :supplier_id, presence: true
+  validates :location, presence: true
   has_attached_file :image, styles: { medium: "300x300>",
                                         thumb: "100x100>" },
                                         default_url: "items/beans.jpg"
```
Comment (bmrsny) [inline app/models/item.rb:10]: do we need validations for unit size and or unit weight.

File: app/views/items/index.html.erb [hunk 6]
```diff
@@ -23,9 +23,11 @@
...
               <p>
-                <%= number_to_currency(item.price.to_f / 100.00) %>
+                <%= number_to_currency(item.format_price) %>
               </p>
-              <%= button_to "Add to Cart", cart_index_path(item_id: item), class: "btn btn-default" %>
+              <% if !current_user || current_user && !current_user.supplier? %>
+                <%= button_to "Add to Cart", cart_index_path(item_id: item), class: "btn btn-default" %>
+              <% end %>
             </div>
           </div>
```
Comment (bmrsny) [inline app/views/items/index.html.erb]: Does this mean there is no add to cart for a guest user?

**Comment by laurawhalin** [comment]
This is nitpicky, but the checkbox labels for shippable and availability would read better as:
This item is currently available
This item is shippable
Otherwise, looks good to me! :snowflake: :snowman: :snowflake:

File: app/views/items/index.html.erb [hunk 7]
```diff
@@ -23,9 +23,11 @@
                 <%= item.title %>
               </h5>
               <p>
-                <%= number_to_currency(item.price.to_f / 100.00) %>
+                <%= number_to_currency(item.format_price) %>
               </p>
-              <%= button_to "Add to Cart", cart_index_path(item_id: item), class: "btn btn-default" %>
+              <% if !current_user || current_user && !current_user.supplier? %>
```
Comment (bmrsny) [inline app/views/items/index.html.erb:28]: Does this mean there is no add to cart button for a guest user?

File: app/views/items/index.html.erb [hunk 8]
```diff
@@ -39,10 +41,16 @@
               			<div class="col-md-8">
                 			<h2><%= item.title %></h2>
                 			<p><%= item.description %></p>
-                		  <p><b>price: <%= number_to_currency(item.price.to_f / 100.00) %></b></p>
-                			<% if !item.retired %>
+                		  <p><b>price: <%= number_to_currency(item.format_price) %></b></p>
+                      <p>Quantity Available: <%= item.quantity %></p>
```
Comment (bmrsny) [inline app/views/items/index.html.erb:45]: I really like this... Really clean

**Pull request merged**